Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/presenter/StatisticsPresenterTest.kt

```diff
@@ -0,0 +1,153 @@
+    }
+
+    @Test
+    fun given_ThePresenterIsPresenting_on_EmissionOfNewLoadingEventWithNoStatistics_it_ShouldNotPresentAnyStatisticsToTheView() {
+        givenThePresenterIsPresenting()
+
+        statisticsEventSubject.onNext(Event.loading<Statistics>())
+
+        Mockito.verify(displayer, never()).display(any(Statistics::class.java))
+    }
+
+    @Test
+    fun given_ThePresenterIsPresenting_on_EmissionOfNewErrorEventWithNoStatistics_it_ShouldNotPresentAnyStatisticsToTheView() {
+        givenThePresenterIsPresenting()
+
+        statisticsEventSubject.onNext(Event.error<Statistics>(SyncError()))
+
+        Mockito.verify(displayer, never()).display(any(Statistics::class.java))
+    }
+
+    @Test
+    fun given_ThePresenterIsNotPresenting_on_EmissionOfNewIdleEventWithStatistics_it_ShouldNotPresentAnythingToTheView() {
+        givenThePresenterIsNotPresenting()
+
+        statisticsEventSubject.onNext(Event.idle<Statistics>().updateData(statistics()))
+
+        Mockito.verifyNoMoreInteractions(displayer)
+    }
+
+    @Test
+    fun given_ThePresenterIsNotPresenting_on_EmissionOfNewLoadingEventWithStatistics_it_ShouldNotPresentAnythingToTheView() {
+        givenThePresenterIsNotPresenting()
+
+        statisticsEventSubject.onNext(Event.loading<Statistics>().updateData(statistics()))
+
+        Mockito.verifyNoMoreInteractions(displayer)
+    }
+
+    @Test
+    fun given_ThePresenterIsNotPresenting_on_EmissionOfNewErrorEventWithStatistics_it_ShouldNotPresentAnythingToTheView() {
+        givenThePresenterIsNotPresenting()
+
+        statisticsEventSubject.onNext(Event.error<Statistics>(SyncError()).updateData(statistics()))
+
+        Mockito.verifyNoMoreInteractions(displayer)
+    }
+
+    @Test
+    fun given_ThePresenterIsNotPresenting_on_EmissionOfNewIdleEventWithNoStatistics_it_ShouldNotPresentAnythingToTheView() {
+        givenThePresenterIsNotPresenting()
+
+        statisticsEventSubject.onNext(Event.idle<Statistics>())
+
+        Mockito.verifyNoMoreInteractions(displayer)
+    }
+
+    @Test
+    fun given_ThePresenterIsNotPresenting_on_EmissionOfNewLoadingEventWithNoStatistics_it_ShouldNotPresentAnythingToTheView() {
+        givenThePresenterIsNotPresenting()
+
+        statisticsEventSubject.onNext(Event.loading<Statistics>())
+
+        Mockito.verifyNoMoreInteractions(displayer)
+    }
+
+    @Test
+    fun given_ThePresenterIsNotPresenting_on_EmissionOfNewErrorEventWithNoStatistics_it_ShouldNotPresentAnythingToTheView() {
+        givenThePresenterIsNotPresenting()
+
+        statisticsEventSubject.onNext(Event.error<Statistics>(SyncError()))
+
+        Mockito.verifyNoMoreInteractions(displayer)
+    }
+
+    private fun givenThePresenterIsPresenting() {
+        presenter.startPresenting()
+    }
+
+    private fun givenThePresenterIsNotPresenting() {
+    }
+
+    private fun statistics() = Statistics.builder().activeTasks(1).completedTasks(3).build()
+
+    private fun setUpService() {
```

Comment:
Maybe re-arrange this private methods in a top-down fashion? Don't mind leaving them like this if it's your standard here.